docs/api/expect.md

```diff
@@ -1405,3 +1405,70 @@ Don't forget to include the ambient declaration file in your `tsconfig.json`.
 :::tip
 If you want to know more, checkout [guide on extending matchers](/guide/extending-matchers).
 :::
+
+## expect.addEqualityTesters <Badge type="info">1.2.0+</Badge>
+
+- **Type:** `(tester: Array<Tester>) => void`
+
+You can use this method to define custom matchers to test if two object equals are equal. It is compatible with Jest's `expect.extend`.
+
+```ts
+class AnagramComparator {
+  public word: string
+
+  constructor(word: string) {
+    this.word = word
+  }
+
+  equals(other: AnagramComparator): boolean {
+    const cleanStr1 = this.word.replace(/ /g, '').toLowerCase()
+    const cleanStr2 = other.word.replace(/ /g, '').toLowerCase()
+
+    const sortedStr1 = cleanStr1.split('').sort().join('')
+    const sortedStr2 = cleanStr2.split('').sort().join('')
+
+    return sortedStr1 === sortedStr2
+  }
+}
+
+function createAnagramComparator(word: string) {
+  return new AnagramComparator(word)
+}
+
+function isAnagramComparator(a: unknown): a is AnagramComparator {
+  return a instanceof AnagramComparator
+}
+
+const areObjectsEqual: Tester = (
+  a: unknown,
+  b: unknown,
+): boolean | undefined => {
+  const isAAnagramComparator = isAnagramComparator(a)
+  const isBAnagramComparator = isAnagramComparator(b)
+
+  if (isAAnagramComparator && isBAnagramComparator)
+    return a.equals(b)
+
+  else if (isAAnagramComparator === isBAnagramComparator)
+    return undefined
+
+  else
+    return false
+}
+
+function* toIterator<T>(array: Array<T>): Iterator<T> {
+  for (const obj of array)
+    yield obj
+}
+
+const customObject1 = createAnagramComparator('listen')
+const customObject2 = createAnagramComparator('silent')
+
+expect.addEqualityTesters([areObjectsEqual])
+```
+
+```ts
+test('objects are equal with different amount', () => {
```

Comment:
The test title feels off? Probably copied from https://jestjs.io/docs/expect#expectaddequalitytesterstesters?

Also I think it's more natural to have objects inside the test as an example:

```ts
test("custom equality tester", () => {
  const customObject1 = createAnagramComparator('listen')
  const customObject2 = createAnagramComparator('silent')
  expect(customObject1).toEqual(customObject2)  
});
```